roles/bmc_fw_update/tasks/main.yml

```diff
@@ -1,2 +1,82 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2022-2024 Dell Inc, or its subsidiaries.
+
 ---
 # tasks file for bmc_fw_update
+
+- name: Get Firmware Inventory
+  ansible.builtin.include_role:
+    name: get_bmc_facts
```

Comment:
get_bmc_facts sounds general, not specific to FW, I assume that later it will just grow which will make it retrive unnecessary information and would result in more and more redfish calls.
I would either:
1. add options to it, specifying what we want to retrieve
2. not separate it to a different role and call FW directly in this role
I prefer [2] since it this is not something that will change in the future we care about a specific FW version

Reply:
- we should separate if we want to re-use it
- we can rename to fw facts to be more specific